### PR TITLE
Add more CI testcases for Exynos Backend

### PR DESCRIPTION
**Pull request by Jiseong-oh**
### Summary
- Support necessary stages for E2E test
- Enable test for float/quant ops and float models

### Test plan

cc @SS-JIA @digantdesai @kimishpatel

### CODE REVIEW

**Comment by SS-JIA** [comment]
@claude help review this PR

**Review by SS-JIA — APPROVED** [review]
Stamp to unblock -- but the agent review comments are valid. Would highly recommend fixing them before landing!

File: backends/samsung/test/models/test_mobilebert_finetuning.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def recover_mobilebert_finetuning(old_func):
+    MobileBertFinetune.load_tokenizer = old_func
+
+
+class Test_Milestone_MobileBertFinetune(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._old_func = patch_mobilebert_finetuning(cls.model_cache_dir)
```
Comment (SS-JIA) [inline backends/samsung/test/models/test_mobilebert_finetuning.py:39]: **High — This new MobileBERT test fails during `setUpClass`.**

`cls.model_cache_dir` is undefined, and `patch_mobilebert_finetuning` accepts no arguments. Even after correcting this call, the replacement loader invokes `AutoTokenizer.from_pretrained` without a required model identifier. Because this test is unskipped and model discovery includes it, a CI run that acquires a Samsung device fails before export.

Please fix the fixture and tokenizer construction and add a host-side setup check that does not require device execution.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/_passes/annotate_qparams.py [hunk 2]
```diff
@@ -148,13 +149,34 @@ def _check_same(requant_obj, ori_obj) -> bool:
...
             ):
-                requantize_map[idx] = requantize_attrs
+                if (
+                    ori_quant_attrs[QuantConstants.QUANT_KEY.quant_dtype]
+                    != requantize_attrs[QuantConstants.QUANT_KEY.quant_dtype]
+                ):
+                    # For Q-DQ who will change quant dtype, we will insert requantization node
+                    requantize_map[idx] = requantize_attrs
+                else:
+                    node.meta["quantize_attrs"] = requantize_attrs
```
Comment (SS-JIA) [inline backends/samsung/_passes/annotate_qparams.py:159]: **High — Same-dtype, different-qparam fan-out loses per-consumer conversion.**

When scale or zero point changes but dtype does not, this overwrites the shared producer's `quantize_attrs` rather than recording edge-specific requantization. With two int8 consumers requiring different qparams, the last branch wins. After Q/DQ folding, another consumer can interpret values with the wrong scale or zero point.

Please retain requantization metadata per consumer for every qparam mismatch, not only dtype changes, and test one producer feeding two same-dtype branches with distinct scales.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: .github/workflows/pull.yml [hunk 3]
```diff
@@ -1558,6 +1558,8 @@ jobs:
         fi
 
         # Test models
+        # TODO: Switch to run_tests.py once CI device provisioning is stable
+        # python -m executorch.backends.samsung.test.utils.run_tests --chipset E9955
```
Comment (SS-JIA) [inline .github/workflows/pull.yml:1565]: **Medium — The new A8W8 operator tests are not run by CI.**

The combined runner remains commented out, while the active command discovers only `backends/samsung/test/models`. No other workflow invokes `backends/samsung/test/ops`, so the roughly 25 new quantized operator cases can regress without signal. This conflicts with the PR's stated CI-expansion goal.

Please enable operator discovery directly or switch to the combined runner once provisioning is stable.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/_passes/transform_quantized_mask.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                break
+        if mask_node is None:
+            return None
+        max_depth = len(nodes_in_pattern)
+        depth = 0
+        while node.target != exir_ops.edge.aten.mul.Tensor:
+            if depth >= max_depth:
+                return None
+            find_next = False
+            for successor in list(node.users.keys()):
```
Comment (SS-JIA) [inline backends/samsung/_passes/transform_quantized_mask.py:49]: **Medium — This is not a safe structural match for the quantized-mask pattern.**

The pass requires the exact placeholder name `attention_mask`, then greedily follows the first successor whose target appears anywhere in an allowed set. It does not validate operator order, consumer target, or operand position before later code assumes one exact MobileBERT topology. Alternate naming, branches, multiple consumers, or reversed add operands can skip the transform or rewrite the wrong nodes. The depth bound prevents an infinite walk but does not validate structure.

Please use an explicit ordered pattern match and verify operand roles before rewriting.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/utils/export_utils.py [hunk 5]
```diff
@@ -78,16 +67,12 @@ def to_edge_transform_and_lower_to_enn(
...
     assert compile_specs is not None, "For now, we must deliver complile specs"
     prog = torch.export.export(module, inputs)
-    pass_list = get_enn_pass_list()
-    if custom_pass_config:
-        pass_list.extend(custom_pass_config)
     return to_edge_transform_and_lower(
         prog,
-        pass_list,
-        {"forward": [EnnPartitioner(compile_specs)]},
+        partitioner={"forward": [EnnPartitioner(compile_specs)]},
```
Comment (SS-JIA) [inline backends/samsung/utils/export_utils.py:76]: **Medium — `custom_pass_config` is accepted but silently ignored.**

The public helper retains the argument, but this call no longer forwards either the former default pass list or caller-provided custom passes to `to_edge_transform_and_lower`. External callers can receive a successful export without their requested transformations.

Please forward `custom_pass_config` through the supported transform-pass argument, or remove/deprecate the parameter explicitly if it is no longer supported.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/test/utils/quant_checkers.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+            quant_out: torch.Tensor = self.current_module(batch_data)
+            _, quant_topk = quant_out.topk(max_topk, dim=-1)
+            quant_topk = quant_topk.t()
+            for k_idx in range(max_topk):
+                correct[k_idx:] += quant_topk[k_idx].eq(fp_top1).view(-1).sum().float()
+        error_messages = []
+        msg_template = "\tFailed in checking Top{}, Target: {:.2f} vs Current: {:.2f}"
+        for topk_num, topk_tol in self.topktol.items():
+            correct_num = correct[topk_num - 1]
+            accuracy_score = correct_num / total * 100
```
Comment (SS-JIA) [inline backends/samsung/test/utils/quant_checkers.py:122]: **Low — Quality checker units and formulas are inconsistent.**

Classifier defaults are `0.9`/`0.95`, but measured agreement is multiplied by 100 before comparison, accepting about 0.9%/0.95% rather than 90%/95%. Segmentation repeats the percent/fraction mismatch. The super-resolution checker also computes `sqrt(sum(square_error)) / num`, which is neither MSE nor RMSE before PSNR calculation. These checks can accept materially broken quantized output.

Please use one unit convention and the standard MSE/PSNR formula, with known-value metric tests.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/_passes/fuse_activation.py [hunk 7]
```diff
@@ -33,45 +33,45 @@ class FuseConvActPass(ExportPass):
...
         self,
         graph_module: GraphModule,
     ):
-        for target_conv, target_act in self.get_target_conv_act(graph_module):
+        for target_src, target_act in self.get_target_src_act(graph_module):
             assert (
                 act_name := self.TARGET_ACTS_MAP.get(target_act.target)(target_act)
             ), f"Not supported {target_act.name} now."
-            target_conv.meta["activation"] = act_name
+            target_src.meta["activation"] = act_name
```
Comment (SS-JIA) [inline backends/samsung/_passes/fuse_activation.py:49]: **Low — Float conv/linear activations are emitted twice.**

The source node receives fused activation metadata before the code checks whether the activation has quant metadata. For float graphs, the `else: continue` leaves the original activation node in place while the source builder also emits the fused activation. Supported activations are idempotent, so output is usually unchanged, but ENN performs redundant work and this pass does not complete its advertised fusion.

Please either erase/rewire the float activation too or delay setting `activation` until the path that will remove the node.

_AI reviewer disclosure: This comment was generated by an AI reviewer._
